### PR TITLE
fix/ fix gateway connectors not showing up in autocomplete

### PR DESCRIPTION
**Pull request by CrimsonJacket**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
Addresses absence of gateway connectors when creating `amm-arb` strategy.

[sc-24477]

**Tests performed by the developer**:

**Tips for QA testing**:
Test that the gateway connectors now appear as part of the autocomplete when creating a `amm-arb` strategy and changing the `connector_1` and `connector_2` configs

### CODE REVIEW

File: hummingbot/strategy/amm_arb/amm_arb_config_map.py
Comment (vic-en) [inline hummingbot/strategy/amm_arb/amm_arb_config_map.py:65]: We do not need to make any change in this file.

File: hummingbot/client/ui/completer.py
Comment (vic-en) [inline hummingbot/client/ui/completer.py:44]: Doing this is going to cause conflict with derivative connector.
The only line that need to be changed in this file is line 39:
```
self._spot_completer = WordCompleter(sorted(AllConnectorSettings.get_exchange_names().union(AllConnectorSettings.get_gateway_evm_amm_connector_names())), ignore_case=True)
```

**Comment by RC-13** [comment]
Confirming that this PR resolved the linked issue, waiting for the comment to be addressed before providing approval.

File: hummingbot/strategy/amm_arb/amm_arb_config_map.py
Comment (CrimsonJacket) [reply hummingbot/strategy/amm_arb/amm_arb_config_map.py:65]: Having "spot" in the prompt is the condition in `_complete_spot_connectors()`. Having "spot" in the prompt will make it s.t. only `_spot_completer` would be used. 
Also, I think using "spot" is not accurate in this case, because the valid connectors include spot exchanges and amm exchanges.

File: hummingbot/client/ui/completer.py
Comment (CrimsonJacket) [reply hummingbot/client/ui/completer.py:44]: I dont think it would be right to include the amm exchanges as a spot connector.

**Review by vic-en — APPROVED** [review]
Changes LGTM

**Pull request merged**
